nomad/leader.go

```diff
@@ -624,68 +624,82 @@ func (s *Server) publishJobSummaryMetrics(stopCh chan struct{}) {
 					break
 				}
 				summary := raw.(*structs.JobSummary)
-				for name, tgSummary := range summary.Summary {
-					if !s.config.DisableTaggedMetrics {
-						labels := []metrics.Label{
-							{
-								Name:  "job",
-								Value: summary.JobID,
-							},
-							{
-								Name:  "task_group",
-								Value: name,
-							},
-						}
-
-						if strings.Contains(summary.JobID, "/dispatch-") {
-							jobInfo := strings.Split(summary.JobID, "/dispatch-")
-							labels = append(labels, metrics.Label{
-								Name:  "parent_id",
-								Value: jobInfo[0],
-							}, metrics.Label{
-								Name:  "dispatch_id",
-								Value: jobInfo[1],
-							})
-						}
-
-						if strings.Contains(summary.JobID, "/periodic-") {
-							jobInfo := strings.Split(summary.JobID, "/periodic-")
-							labels = append(labels, metrics.Label{
-								Name:  "parent_id",
-								Value: jobInfo[0],
-							}, metrics.Label{
-								Name:  "periodic_id",
-								Value: jobInfo[1],
-							})
-						}
-
-						metrics.SetGaugeWithLabels([]string{"nomad", "job_summary", "queued"},
-							float32(tgSummary.Queued), labels)
-						metrics.SetGaugeWithLabels([]string{"nomad", "job_summary", "complete"},
-							float32(tgSummary.Complete), labels)
-						metrics.SetGaugeWithLabels([]string{"nomad", "job_summary", "failed"},
-							float32(tgSummary.Failed), labels)
-						metrics.SetGaugeWithLabels([]string{"nomad", "job_summary", "running"},
-							float32(tgSummary.Running), labels)
-						metrics.SetGaugeWithLabels([]string{"nomad", "job_summary", "starting"},
-							float32(tgSummary.Starting), labels)
-						metrics.SetGaugeWithLabels([]string{"nomad", "job_summary", "lost"},
-							float32(tgSummary.Lost), labels)
+				if s.config.DisableDispatchedJobSummaryMetrics {
+					job, err := state.JobByID(ws, summary.Namespace, summary.JobID)
+					if err != nil {
+						s.logger.Printf("[ERR] nomad: failed to lookup job for summary: %v", err)
```

Comment:
This broke the build:
```suggestion
						s.logger.Error("error getting job for summary", "error", err)
```